alica_engine/include/engine/blackboard/Blackboard.h

```diff
@@ -308,22 +313,30 @@ class BlackboardImpl
         };
     };
 
-    // trait to check if a type is one of the types in the variant
-    template <class T1, class T2>
-    struct isTypeInVariant;
+    // trait to find the index of the given type in the variant, -1 if not found
+    template <class T, class Variant, std::size_t numTypes>
+    struct findInVariantImpl;
 
-    template <class T>
-    struct isTypeInVariant<T, std::variant<>> : std::false_type
+    template <class T, std::size_t NUM_TYPES>
+    struct findInVariantImpl<T, std::variant<>, NUM_TYPES> : std::integral_constant<std::ptrdiff_t, -1>
     {
     };
 
-    template <class T, class... VariantTs>
-    struct isTypeInVariant<T, std::variant<T, VariantTs...>> : std::true_type
+    template <class T, class... VariantTs, std::size_t NUM_TYPES>
+    struct findInVariantImpl<T, std::variant<T, VariantTs...>, NUM_TYPES> : std::integral_constant<std::ptrdiff_t, NUM_TYPES - sizeof...(VariantTs) - 1>
```

Comment:
it could be a huge improvement if we could get rid of `std::monostate` in our variant: not need to make this -1
all variant types has a default ctor, moreover, we just need only first type could be default constructed CMIIW

Reply:
not sure that this execution patch is really used... but yeah, it's out of scope

Reply:
yes it is used